### ISSUE TITLE
Badips tries to unban when stopping

### ISSUE THREAD

**Issue by benrubson** [fail2ban/fail2ban#2602]
Hi,

Some firewalls, such as iptables, bsd-ipfw... have a stop action to flush chains when Fail2Ban stops.
However then, Badips tries to unban every banned IP, leading to the following log :

```
2020-01-15 19:19:25,578 fail2ban.actions        [18862]: NOTICE  [badips] Flush ticket(s) with iptables-allports
2020-01-15 19:19:25,677 fail2ban.utils          [18862]: #39-Lev. 7f02a1d32298 -- exec: iptables -w -D f2b-badips -s A.B.C.D -j DROP
2020-01-15 19:19:25,678 fail2ban.utils          [18862]: ERROR   7f02a1d32298 -- stderr: 'iptables: Bad rule (does a matching rule exist in that chain?).'
2020-01-15 19:19:25,678 fail2ban.utils          [18862]: ERROR   7f02a1d32298 -- returned 1
(...)
```

I assume this is due to the following `badips.py` code :
```
        def stop(self):
                """If `banaction` set, clears blacklisted IPs.
                """
                if self.banaction is not None:
                        if self._timer:
                                self._timer.cancel()
                                self._timer = None
                        self._unbanIPs(self._bannedips.copy())
```

But then I'm not sure how to keep it compatible with both firewalls having and not having stop action.

Thank you 👍

**Comment by sebres** [comment]
You'll catch the same "issue" if some new features like bulk-ban (and -unban) get introduced,
Initially badips was implemented as many capabilities like `actionflash`, `actionrepair` etc were not yet introduced, so it should recognize it and provide a new wrapper.

BTW, this happens because at the moment it is built like a wrapper around real action, where I think it could be better where a real action would call badips functionality (second action or some new callbacks).
Basically every feature a real action may get could bother badips (and vice versa), for example how it built now, it will be also unable to restore sane environment after failed invariant check (see #2588).
So I think the whole concept of badips-action should get a fresh view.

**Comment by benrubson** [comment]
Understood @sebres, many thanks for your detailed answer.
`badips.py` then suffers from some limitations, and could get worst in the future.
Time for a fresh view certainly yes.
Or (and / or ?) for the network feature discussed in #881.

**Comment by benrubson** [comment]
#2603 solves this 👍

**Comment by benrubson** [comment]
"Strangely", as originally stated, I face these error messages with **Fail2Ban v0.10.2, Debian 10.2** :
```
2020-01-15 19:19:25,578 fail2ban.actions        [18862]: NOTICE  [badips] Flush ticket(s) with iptables-allports
2020-01-15 19:19:25,677 fail2ban.utils          [18862]: #39-Lev. 7f02a1d32298 -- exec: iptables -w -D f2b-badips -s A.B.C.D -j DROP
2020-01-15 19:19:25,678 fail2ban.utils          [18862]: ERROR   7f02a1d32298 -- stderr: 'iptables: Bad rule (does a matching rule exist in that chain?).'
2020-01-15 19:19:25,678 fail2ban.utils          [18862]: ERROR   7f02a1d32298 -- returned 1
(...)
```

But I did not face them with **Fail2Ban v0.10.5, CentOS 7.7** :
```
2020-02-10 18:41:24,948 fail2ban.actions        [3616]: NOTICE  [badips] Flush ticket(s) with iptables-allports
2020-02-10 18:41:24,960 fail2ban.BadIPsAction   [3616]: INFO    Unbanned IP A.B.C.D for jail 'badips' with action 'iptables-allports'
(...)
```
IPs are individually proceeded, even if they have been flushed just before, but this does not trigger any error message.

Something may have changed between these 2 versions to avoid triggering an error message on such cases...
